cpp/include/raft/spatial/knn/detail/fused_l2_knn.cuh

```diff
@@ -300,6 +327,16 @@ __global__ __launch_bounds__(Policy::Nthreads, 2) void fusedL2kNN(
     [numOfNN, sqrt, m, n, ldd, out_dists, out_inds] __device__(
       AccT acc[Policy::AccRowsPerTh][Policy::AccColsPerTh], DataT * regxn,
       DataT * regyn, IdxT gridStrideX, IdxT gridStrideY) {
+      if (useNorms) {
+#pragma unroll
```

Comment:
Aside from the template instantiations for each distance, is it possible that the unrolling of these loops could also be having a drastic increase on the compile time?